### DISCUSSION TITLE
Status Update: 2022-10-25

### DISCUSSION THREAD

**Discussion by haixuanTao** [dora-rs/dora#113] [Status Update]
Status update :smile_cat:

**Comment by haixuanTao** [comment]
## "dora-rover"

- Made ROS-based `ndt` work on rover
- Connect ROS to dora through Python API
- Test rover control through `mavlink`. Some issues persist with mavlink not exposing a steering API.
- Installed `dora` on rover Nvidia AGX Xavier. 
- Need to recompile `pytorch CUDA` on AGX Xavier as it is ARM-based CUDA GPU which is very buggy in terms of dependencies.
-> Anything that requires GPU on the rover will require to be recompiled for ARM CUDA.

Next step is going to be to port `dora-drives` into "`dora-rover`"

**Comment by haixuanTao** [comment]
## Robotic Club Sponsoring

- Began discussion with CentraleSupelec Robotic club for a potential sponsoring. Agreed for a `dora` and `dora-drives` presentation.
- Potential pilot project for growing European community.

**Comment by phil-opp** [comment]
## Command-line Interface

We merged the following updates to the dora CLI in the past two weeks:

- Assign each started dataflow a UUID
- Allow stopping running dataflows `dora stop <UUID>`
  - Sends a new stop message to all operators and nodes
  - Instructs the operators/nodes to stop immediately, without processing further input
  - See [#105](https://github.com/dora-rs/dora/pull/105) for details
- The `dora destroy` command now stops all running dataflows before stopping the `dora-coordinator`
- Implement `dora list` command to list running dataflows ([#107](https://github.com/dora-rs/dora/pull/107))

Our future plans for the CLI are tracked in [#93](https://github.com/dora-rs/dora/issues/93). The following commands are planned for the `v0.1` release:

- `dora check` without arguments to check if `dora-coordinator` and `iceoryx` daemon are running
- `dora up` for starting all required daemons that are not running yet
  - requires that all daemon executables are in the `PATH`
- `dora add` templates for Python and for `dataflow.yml` files

**Reply by haixuanTao** [reply]
Lifecycle:
```
dora up -> dora start -> dora stop -> dora destroy
```

**Comment by phil-opp** [comment]
## Refer to operators/nodes through URL

In [#108](https://github.com/dora-rs/dora/pull/108), we implement support for downloading compiled operators/nodes from remote locations. For example, the following dataflow uses a GitHub release artifact as a custom node:

```yaml
nodes:
  - id: rust-node
    custom:
      source: https://github.com/dora-rs/dora/releases/download/v0.0.0-test.4/rust-dataflow-example-node
      inputs:
        tick: dora/timer/millis/300
      outputs:
        - random
```
